### PR TITLE
fix(curriculum): HTML-challenges: links are inconsistent.

### LINKED ISSUE #41886: HTML-challenges: links are inconsistent.
**Describe your problem and how to reproduce it:**
About **19** of the HTML lessons contain links, either in the html-code or in the description&instructions.

**16** of them use short links. (for example `freecodecamp.org`)
**2** of them use long links. (for example `www.freecodecamp.org`)
**1** uses a mixture of both.

I suggest using one style for all challenges.
Personally, I'm in favor of the shorter variant, as it's easier to read for new students.

**Add a Link to the page with the problem:**
This originated here: https://github.com/freeCodeCamp/freeCodeCamp/pull/41876#issuecomment-824105202

**Links to challenges:**
`thisisalink.com`
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/link-to-external-pages-with-anchor-elements
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/link-to-internal-sections-of-a-page-with-anchor-elements
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/make-dead-links-using-the-hash-symbol
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/turn-an-image-into-a-link
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/create-a-bulleted-unordered-list
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/create-an-ordered-list
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/create-a-text-field
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/add-placeholder-text-to-a-text-field
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/create-a-form-element
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/use-html5-to-require-a-field
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/create-a-set-of-radio-buttons
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/create-a-set-of-checkboxes
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/check-radio-buttons-and-checkboxes-by-default
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/nest-many-elements-within-a-single-div-element

`www.thisisalink.com`
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/link-to-external-pages-with-anchor-elements
1. https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/nest-an-anchor-element-within-a-paragraph

Mixture of both:
https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/add-images-to-your-website

**Tell us about your browser and operating system:**
* Browser Name: Firefox
* Browser Version: 88.0 (64-bit)
* Operating System: Linux

### PR DESCRIPTION
**Pull request by jakobnarobe**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #41886

Updated links in Html- challenges from ```thisislink.com``` to ```www.thisislink.com``` so they are consistent.
Updated regex that broke with the first commit.

### CODE REVIEW

**Comment by EboranBokaj** [comment]
Hi, I hope everything is as it should be. This is my first PR so if there is anything I need to change please let me know, I appreciate the feedback. Thanks!

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ assert($('img').length);
 Your image should have a `src` attribute that points to the kitten image.
 
 ```js
-assert(/^https:\/\/bit\.ly\/fcc-relaxing-cat$/i.test($('img').attr('src')));
+assert(/^https:\/\/www.bit\.ly\/fcc-relaxing-cat$/i.test($('img').attr('src')));
```
Comment (ieahleen) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.md:55]: the dot needs to be escaped
```suggestion
assert(/^https:\/\/www\.bit\.ly\/fcc-relaxing-cat$/i.test($('img').attr('src')));
```

File: curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/link-to-external-pages-with-anchor-elements.md [hunk 2]
```diff
@@ -31,10 +31,10 @@ Your `a` element should have the anchor text of `cat photos`.
 assert(/cat photos/gi.test($('a').text()));
 ```
 
-You need an `a` element that links to `https://freecatphotoapp.com`
+You need an `a` element that links to `https://www.freecatphotoapp.com`
 
 ```js
-assert(/^https?:\/\/freecatphotoapp\.com\/?$/i.test($('a').attr('href')));
+assert(/^https?:\/\/www.freecatphotoapp\.com\/?$/i.test($('a').attr('href')));
```
Comment (ieahleen) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/link-to-external-pages-with-anchor-elements.md:37]:
```suggestion
assert(/^https?:\/\/www\.freecatphotoapp\.com\/?$/i.test($('a').attr('href')));
```

**Review by gikf — CHANGES_REQUESTED** [review]
Looking at the list of files in issue, there's one missing in PR.

**Comment by gikf** [comment]
@EboranBokaj for future - it's best to make changes on a separate branch, not the `main` branch of the fork. Take look at point 3 and further at https://contribute.freecodecamp.org/#/how-to-setup-freecodecamp-locally?id=making-changes-locally

**Comment by raisedadead** [comment]
> Looking at the list of files in issue, there's one missing in PR.

Hey @gikf thanks for the QA. Do you mind letting us know which one? Thanks.

**Comment by gikf** [comment]
This one https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/nest-an-anchor-element-within-a-paragraph.md

**Comment by raisedadead** [comment]
Hi @EboranBokaj can update the one missed file as shared by @gikf above. This is almost good to go. Thanks.

**Comment by EboranBokaj** [comment]
Hi, I will update it now, sorry I had a lot of work during the weekend...

**Comment by EboranBokaj** [comment]
Thanks for the feedback @gikf, I will make sure to use a new branch the next time!

**Review by gikf — APPROVED** [review]

**Review by raisedadead — APPROVED** [review]

**Comment by raisedadead** [comment]
#### :sparkles: :sparkles: **AWESOME! LGTM.** :sparkles: :sparkles: 

Thanks @EboranBokaj, for this pull request and contributing to the code-base for the first time.

We will be looking forward for more contributions in future.

Cheers and happy contributing!

**Pull request merged**